elements/lisk-p2p/src/p2p.ts

```diff
@@ -182,6 +182,14 @@ const createPeerPoolConfig = (
 	peerLists,
 });
 
+const filteradvertiseAddressPeer = (peer: P2PPeerInfo) => {
+	if (peer.internalState && !peer.internalState.advertiseAddress) {
```

Comment:
Can we change this condition to `peer.internalState.advertiseAddress === false` because in runtime if someone doesn't send `advertiseAddress` it could be `undefined` and still return false, i.e., it will not advertise the address who doesn't explicitly mention it to be advertise.